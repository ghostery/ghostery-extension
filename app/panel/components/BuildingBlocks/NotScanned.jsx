/**
 * Not Scanned Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';

/**
 * @class Implements the Not Scanned component displayed in the Summary view
 * when a site is not scannable or has not yet been scanned.
 * @memberof PanelBuildingBlocks
 */
const NotScanned = ({ isSmall }) => {
	const notScannedClassNames = ClassNames('sub-component', 'not-scanned', {
		small: isSmall,
	});

	return (
		<div className={notScannedClassNames}>
			<div className="not-scanned-header">
				{t('summary_page_not_scanned') }
			</div>
			<div className="not-scanned-text">
				{ t('summary_description_not_scanned_1') }
			</div>
			<div className="not-scanned-text">
				{ t('summary_description_not_scanned_2') }
			</div>
		</div>
	);
};

NotScanned.propTypes = {
	isSmall: PropTypes.bool,
};

NotScanned.defaultProps = {
	isSmall: false,
};

export default NotScanned;
