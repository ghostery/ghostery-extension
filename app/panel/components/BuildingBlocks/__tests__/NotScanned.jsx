/**
 * Not Scanned Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import NotScanned from '../NotScanned';

// Fake the translation function to only return the translation key
global.t = function(str) {
	return str;
};

describe('app/panel/components/NotScanned.jsx', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('NotScanned is rendered correctly when no props are passed', () => {
			const component = renderer.create(
				<NotScanned />
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('NotScanned is rendered correctly when small', () => {
			const component = renderer.create(
				<NotScanned isSmall />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});
});
