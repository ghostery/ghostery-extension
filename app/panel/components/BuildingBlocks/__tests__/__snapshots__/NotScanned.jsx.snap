// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/panel/components/NotScanned.jsx Snapshot tests with react-test-renderer NotScanned is rendered correctly when no props are passed 1`] = `
<div
  className="sub-component not-scanned"
>
  <div
    className="not-scanned-header"
  >
    summary_page_not_scanned
  </div>
  <div
    className="not-scanned-text"
  >
    summary_description_not_scanned_1
  </div>
  <div
    className="not-scanned-text"
  >
    summary_description_not_scanned_2
  </div>
</div>
`;

exports[`app/panel/components/NotScanned.jsx Snapshot tests with react-test-renderer NotScanned is rendered correctly when small 1`] = `
<div
  className="sub-component not-scanned small"
>
  <div
    className="not-scanned-header"
  >
    summary_page_not_scanned
  </div>
  <div
    className="not-scanned-text"
  >
    summary_description_not_scanned_1
  </div>
  <div
    className="not-scanned-text"
  >
    summary_description_not_scanned_2
  </div>
</div>
`;
